### ISSUE TITLE
"Cannot use 'in' operator to search for 'name' in null" during Amplify Codegen @aws-amplify/cli @12.7.0

### ISSUE THREAD

**Issue by ObinnaAka** [aws-amplify/amplify-cli#13338]
### How did you install the Amplify CLI?

pnpm

### If applicable, what version of Node.js are you using?

20.7.0

### Amplify CLI Version

12.7.0

### What operating system are you using?

Ma

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes made

### Describe the bug

When running amplify codegen, i get the following output

`amplify codegen             
✔ Generated GraphQL operations successfully and saved at graphql
✖ Cannot use 'in' operator to search for 'name' in null`

When I downgrade to @aws-amplify/cli@11.1.1 it works

### Expected behavior

`✔ Generated GraphQL operations successfully and saved at graphql
✔ Code generated successfully and saved in file api/API.ts`

### Reproduction steps

Run amplify codegen. I cannot identify if any change to the schema led to this.

### Project Identifier

a1fad6e952abc1a3b581c4d3442702d2

### Log output

<details>

```
# Put your logs below this line

✖ Cannot use 'in' operator to search for 'name' in null
```

</details>

### Additional information

After downgrading to v11.1.1 then upgrading back to 12.7.0 it now works as intended.

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by ykethan** [comment]
Hey @ObinnaAka, thank you for reaching out. The issue appears to be similar to https://github.com/aws-amplify/amplify-codegen/issues/734. additionally, a fix has been merged here:https://github.com/aws-amplify/amplify-codegen/pull/738 and should be released in the upcoming updates.

A workaround on the comment https://github.com/aws-amplify/amplify-codegen/issues/734#issuecomment-1757743003 seems to mitigate the issue; which suggests adding `API.ts` as an exclusion in the `.graphqlconfig.yml` file.

**Comment by jgo80** [comment]
Same for me. I fear downgrading to v11 will kill my project. Please fix 12.7 🙏

**Comment by ykethan** [comment]
Closing as the issue as the PR has been merged. please feel free in reaching out to us if the issue reoccurs.
